scripts/taxonomies/lint_taxonomy.pl

```diff
@@ -0,0 +1,508 @@
+		my %props = ();    # lines defining properties
+		my @original_lines = ();
+		# non meaningful lines above a meaningful line (entry, parent or porperty)
+		my @previous_lines = ();
+		my @errors = ();
+		my $entry_start_line = $line_num + 1;
+		while (my $line = $lines_iter->()) {
+			$line_num += 1;
+			push @original_lines, $line;    # collect lines for comparison
+
+			# blank line means we are changing entry, so let's return collected data
+			if ($line =~ /^\s*$/) {
+				my $entry = {
+					parents => \@parents,
+					entry_id_line => $entry_id_line,
+					entries => \%entries,
+					props => \%props,
+					original_lines => \@original_lines,
+					tail_lines => \@previous_lines,
+					start_line => $entry_start_line,
+					end_line => $line_num,
+					errors => \@errors,
+				};
+				add_entry_id($entry, \@errors);
+				# return $entry
+				return $entry;
+			}
+			# parents
+			elsif ($line =~ /^</) {
+				# if we already have an entry id, parent are not at the good position,
+				# and it might mean two entries where merged inadvertantly (eg. by an auto merge)
+				if ($entry_id_line) {
+					push(
+						@errors,
+						{
+							severity => "Error",
+							type => "Correctness",
+							line => $line_num,
+							message => (
+									  "Parent in the middle of an entry, might mean erroneous merge of two entries:\n"
+									. "- $line"
+							)
+						}
+					);
+				}
+				push @parents, {line => $line, previous => [@previous_lines], line_num => $line_num};
+				@previous_lines = ();
+			}
+			# synonym
+			elsif ($line =~ /^(\w+):[^:]*(,.*)*$/) {
+				if (!defined $entry_id_line) {
+					$entry_id_line = {line => $line, previous => [@previous_lines], lc => $1,, line_num => $line_num};
+				}
+				else {
+					my $lc = $1;
+					if ((defined $entries{$lc}) || ($entry_id_line->{lc} eq $lc)) {
+						my $previous_lc_line;
+						if (defined $entries{$lc}) {
+							$previous_lc_line = $entries{$lc}{line};
+						}
+						else {
+							$previous_lc_line = $entry_id_line->{line};
+						}
+
+						push @errors,
+							{
+							severity => "Error",
+							type => "Correctness",
+							line => $line_num,
+							message => ("duplicate synonym for $lc:\n" . "- $previous_lc_line" . "- $line")
+							};
+					}
+					# but try to do our best and continue
+					if (defined $entries{$lc}) {
+						$entries{$lc}{line} = $entries{$lc}{line} . $line;
+						push @{$entries{$lc}{previous}}, @previous_lines;
+					}
+					else {
+						$entries{$lc} = {line => $line, previous => [@previous_lines],, line_num => $line_num};
+					}
+				}
+				@previous_lines = ();
+			}
+			# property
+			elsif ($line =~ /^(\w+):(\w+):(.*)$/) {
+				my $prop = $1;
+				my $lc = $2;
+				if (defined $props{"$prop:$lc"}) {
+					push(
+						@errors,
+						{
+							severity => "Error",
+							type => "Correctness",
+							line => $line_num,
+							message => (
+									  "duplicate property value for $prop:$lc:\n" . "- "
```

Comment:
Suggestion to make it more clear that it's not the value that is duplicate, but that we have 2 lines for the same property in the same language

```suggestion
									  "duplicate property language line for $prop:$lc:\n" . "- "
```